In [122]:
import pandas as pd
import numpy as np

### 1)
Провести дисперсионный анализ для определения того, есть ли различия среднего роста среди взрослых футболистов, хоккеистов и штангистов. Даны значения роста в трех группах случайно выбранных спортсменов: Футболисты: 173, 175, 180, 178, 177, 185, 183, 182. Хоккеисты: 177, 179, 180, 188, 177, 172, 171, 184, 180. Штангисты: 172, 173, 169, 177, 166, 180, 178, 177, 172, 166, 170.

In [125]:
n1 = 8
n2 = 9
n3 = 11
y1 = np.array([173, 175, 180, 178, 177, 185, 183, 182])
y2 = np.array([177, 179, 180, 188, 177, 172, 171, 184, 180])
y3 = np.array([172, 173, 169, 177, 166, 180, 178, 177, 172, 166, 170])

In [126]:
Y1 = y1.sum()/n1 # Футболисты
Y2 = y2.sum()/n2 # Хоккеисты
Y3 = y3.sum()/n3 # Штангисты

In [100]:
Y = (sum(y1)+sum(y2)+sum(y3))/(n1+n2+n3)

In [108]:
Y_all = np.hstack((y1,y2,y3))
d = []
for i in Y_all:
    s_err = (i - Y)**2
    d.append(s_err)
S = sum(d)

In [109]:
Sf = ((Y1-Y)**2)*8 + ((Y2-Y)**2)*9 + ((Y3-Y)**2)*11

In [110]:
a = []
for i in y1:
    s_err = (i - Y1)**2
    a.append(s_err)
for i in y2:
    s_err = (i - Y2)**2
    a.append(s_err)
for i in y3:
    s_err = (i - Y3)**2
    a.append(s_err)
S_ost = sum(a)

In [111]:
S = Sf + S_ost

In [112]:
σ = S / ((n1+n2+n3)-1)
σ_f = Sf / (3-1)
σ_ost = S_ost / (28-3)

In [116]:
Fh = σ_f/σ_ost
Fкрит = 3.38

In [120]:
print("среднее значение роста среди Футболистов  = ", Y1)
print("среднее значение роста среди Хоккеистов  = ", Y2)
print("среднее значение роста среди Штангистов  = ", Y3)
print("среднее значение роста по всем = ", Y)
print("Сумма квадратов отклонений наблюдений от общего среднего  = ", S)
print("Сумма квадратов отклонений средних групповых значений от общего среднего значения Y = ", Sf)
print("Остаточная сумма квадратов отклонений = ", S_ost)
print("Общая дисперсия = ", σ)
print("Факторная дисперсия = ", σ_f)
print("Остаточная дисперсия = ", σ_ost)
print("Наблюдаемое значение = ", Fh)
print("Находим Fкрит в таблице распределения Фишера Снедекора для уровня значимости α=0.05 и двух степений свободы = ", Fкрит)

среднее значение роста среди Футболистов  =  179.125
среднее значение роста среди Хоккеистов  =  178.66666666666666
среднее значение роста среди Штангистов  =  172.72727272727272
среднее значение роста по всем =  176.46428571428572
Сумма квадратов отклонений наблюдений от общего среднего  =  830.9642857142859
Сумма квадратов отклонений средних групповых значений от общего среднего значения Y =  253.9074675324678
Остаточная сумма квадратов отклонений =  577.0568181818181
Общая дисперсия =  30.776455026455032
Факторная дисперсия =  126.9537337662339
Остаточная дисперсия =  23.082272727272724
Наблюдаемое значение =  5.500053450812599
Находим Fкрит в таблице распределения Фишера Снедекора для уровня значимости α=0.05 и двух степений свободы =  3.38


Ответ: Fh > Fкрит = 5.50 > 3.38 Следовательно, различие между группами статистически значимое 